File: 2º/PDOO/ProyectoRUBY/lib/dado.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require 'singleton'
#require_relative "diario.rb"
module Civitas
  class Dado
    include Singleton
    def tirar
      if(@debug==false)
        @random=rand(6)+1
        @random=@random.to_i
        @ultimoResultado=1+@random
        return @ultimoResultado
      else
        @ultimoResultado=1
        return @ultimoResultado
      end
    end
    def salgoDeLaCarcel
      if(tirar>=5)
        return true
      else
        return false
      end
    end
    def quienEmpieza(n)
      @random=rand(n)
      @random=@random.to_i
      return @random
    end
    def setDebug(d)
      @debug=d
      e=@debug==true ? "Debug->true":"Debug->false"
      Diario.instance.ocurre_evento(e)
    end
    attr_reader :ultimoResultado, :debug, :random, :SalidaCarcel, :instance
    private :debug, :random, :ultimoResultado
    private :SalidaCarcel, :instance
    private
    def initialize()
      @random=rand()
      @ultimoResultado=1
      @debug=true
      @SalidaCarcel=5
    end
  end
end

File: 2º/PDOO/ProyectoJAVA/src/civitas/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.Random;

/**
 *
 * @author nikitastetskiy
 */
public class Dado {

  static final private Dado instance = new Dado();
  static final private int SalidaCarcel = 5;
  
  Random random;
  int ultimoResultado;
  boolean debug;

  private Dado(){
      ultimoResultado=0;
      random= new Random();
      debug=false;
  }
    
  int getUltimoResultado(){
      return ultimoResultado;
  }
  
  public static Dado getInstance() {
      return instance;
  }
    
  int tirar (){
      if(debug){
          return ultimoResultado= 1;
      }
      else{
          return ultimoResultado= 1+random.nextInt(6);
      }
  }
  
  boolean salgoDeLaCarcel(){
      int n = tirar();
      boolean x = n >= 5 ? true : false;
      return x;
  }
  
  public int quienEmpieza(int n){
      return random.nextInt(n);
  }
  
  public void setDebug(Boolean d){
      debug = d;
      String e= debug == true ? "Debug-> true":"Debug-> false";
      Diario.getInstance().ocurreEvento(e);
  }

}
